### PR TITLE
README and LICENSE

### PR DESCRIPTION
**Pull request by WillemVervoort**
### All Submissions:

* [x ] Have you followed the guidelines in our **CONTRIBUTING** document?
* [ x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue. List the bug ID if applicable)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing core functionality to change) 

### New Feature Submissions checklist:

1. [x] Have you ensured there are no conflicts with major packages (e.g. tidyverse, MASS) or base R packages?
4. [x] Have you linted your code with lintr locally prior to submission?
5. [ ] Have you styled your code with styler locally prior to submission?

### Changes to Core Features:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them? see Issue #54
* [ ] Have you added tests for your core features and do your core features pass all tests?
* [ ] Have you successfully run tests with your changes locally?
* [ ] Does your code run on all major platforms (Windows, macOS, Linux)?
* [ ] Have you demonstrated backwards compatibility and compatibility with the current development version, or discussed a potential break in backwards compatibility?
* [ ] Have you updated the package vignettes? 

## What is this PR about

Issue #54 Create a README

Note that as part of this I have added a license as the standard README template has it as "required".
Template used: [https://github.com/RichardLitt/standard-readme/blob/master/spec.md](https://github.com/RichardLitt/standard-readme/blob/master/spec.md)

### CODE REVIEW

File: LICENSE.txt [hunk 1]
```diff
@@ -0,0 +1,21 @@
+MIT License
+
+Copyright (c) [2020] [Joseph Guillaume; Felix Andrews]
```
Comment (jnothman) [inline LICENSE.txt:3]:
```suggestion
Copyright (c) 2020, Joseph Guillaume and Felix Andrews.
```

This would be more conventional punctuation?

Comment (josephguillaume) [reply LICENSE.txt:3]: License is actually GPL>=2 https://github.com/josephguillaume/hydromad/blob/master/DESCRIPTION#L17
Github also has template licenses that can be applied: https://help.github.com/en/github/building-a-strong-community/adding-a-license-to-a-repository
I can add that if this PR removes this file?

File: README.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+--->
+
+Currently you can install the development version from
+[GitHub](https://github.com/) with:
+
+``` r
+# install.packages("devtools")
+devtools::install_github("JosephGuillaume/hydromad")
+```
+
```
Comment (josephguillaume) [inline README.md:38]: These instructions don't match those on the website. 
http://hydromad.catchment.org/#installation
https://github.com/josephguillaume/hydromad/blob/master/web/template.html#L140
I think we still want to encourage installation of the compiled version, which doesn't require installation of Rtools

I think what we want is actually to simplify the instructions of the website and here to add hydromad.catchment.org to repos, i.e. `options(repos = r)`. That'll make it easier to deal with dependencies.

File: README.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+#> scale    NA    NA     
+#> l         0     0 (==)
+#> p         1     1 (==)
+#> t_ref    20    20 (==)
+#> Routing Parameters:
+#>       lower upper  
+#> tau_s     2   100  
+#> v_s       0     1
+```
+
```
Comment (josephguillaume) [inline README.md:83]: Example should include a `fitBy` method (`fitByOptim`?) and `summary` I think?
We should aim for the example to capture the philosophy of how hydromad approaches interaction with models?

**Comment by josephguillaume** [comment]
There seems to be an extra png that isn't used?
Should we perhaps just include a static md rather than run the example dynamically?

File: README.Rmd [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  collapse = TRUE,
+  comment = "#>",
+  fig.path = "man/figures/README-",
+  out.width = "100%"
+)
+```
+
+# hydromad
+
+<!-- badges: start -->
```
Comment (jnothman) [inline README.Rmd:18]: Can apply [badgecreatr](https://cran.r-project.org/web/packages/badgecreatr/readme/README.html) if you like... May make more sense after we have #53 merged

File: LICENSE.txt (hunk 1 above)
Comment (WillemVervoort) [reply LICENSE.txt:3]: I can also change the license.txt? What do you prefer? I used the suggested github link, which where I came out to the MIT, but I didn't check your description :-(

File: README.md (hunk 2 above)
Comment (WillemVervoort) [reply README.md:38]: OK I will update to the same as the website and also make the link to the website more explicit (I forgot that)
Do we want people to install from github at all? 

"here to add hydromad.catchment.org to repos": I am not sure I know how to do that

File: README.md (hunk 3 above)
Comment (WillemVervoort) [reply README.md:83]: Sure, I just wanted to keep it small. I wasn't totally sure what a good example was, in fact you probably have a better idea. Please suggest and I will include

**Comment by WillemVervoort** [comment]
> There seems to be an extra png that isn't used?
> Should we perhaps just include a static md rather than run the example dynamically?

Hmm, yes, I think I copied from the website initially to include the image with the "philosophy" but forgot to load the png. What is your thought about this? Include?

File: README.Rmd (hunk 4 above)
Comment (WillemVervoort) [reply README.Rmd:18]: Indeed, I was waiting for that, which is why I left it commented out

**Comment by WillemVervoort** [comment]
> Ideally we should have a workflow item that builds the .md from the .Rmd, or checks that they are in sync??

The package "usethis" did not make any suggestions about this. I just use that package and the function based on your @jnothman suggestion. I guess that is an issue, "usethis" assumes that you set up your package in R-studio and interact with github that way. If we don't do that then yes, the .md and .rmd can get disconnected. I used the .Rmd  because it was easier to include the Rcode example that was suggested in the template README

File: LICENSE.txt (hunk 1 above)
Comment (WillemVervoort) [reply LICENSE.txt]: I have now removed LICENSE.txt and pointed to DESCRIPTION

**Comment by mortisflux** [comment]
> > Ideally we should have a workflow item that builds the .md from the .Rmd, or checks that they are in sync??
> 
> The package "usethis" did not make any suggestions about this. I just use that package and the function based on your @jnothman suggestion. I guess that is an issue, "usethis" assumes that you set up your package in R-studio and interact with github that way. If we don't do that then yes, the .md and .rmd can get disconnected. I used the .Rmd because it was easier to include the Rcode example that was suggested in the template README

I think we can use [this](https://fromthebottomoftheheap.net/2020/04/30/rendering-your-readme-with-github-actions/) as a basic workflow to generate the md from the Rmd.

**Comment by WillemVervoort** [comment]
Yes, that looks straight forward enough, I will try that and add that to the PR

**Comment by WillemVervoort** [comment]
> Yes, that looks straight forward enough, I will try that and add that to the PR
> 
> Added this, but not sure how I test

Aah, this happens with every new commit. It it not as easy as I thought as the install.packages("hydromad", repos ="http://hydromad.catchment.org") fails on ppso and dream packages..... Of course!!

**Comment by WillemVervoort** [comment]
OK, removed the R-CMD check @josephguillaume can you look at the other comments and replies and what you want there?

File: .Rbuildignore [hunk 5]
```diff
@@ -2,3 +2,5 @@ R/ltv.R
 inst/doc/hydromad.Rnw
 inst/doc/runlists.Rnw
 inst/doc/dataChecking.Rnw
+^README\.Rmd$
```
Comment (josephguillaume) [inline .Rbuildignore:5]:
```suggestion
^README\.Rmd$
^README\.md$
```

File: .Rhistory [hunk 6]
```diff
@@ -0,0 +1,3 @@
+require(usethis)
```
Comment (josephguillaume) [inline .Rhistory:1]: Remove this file and add .Rhistory to .gitignore

File: README.md (hunk 3 above)
Comment (josephguillaume) [reply README.md:122]: Ok for now. Thanks!

**Comment by josephguillaume** [comment]
> OK, removed the R-CMD check @josephguillaume can you look at the other comments and replies and what you want there?

Thanks @WillemVervoort ! I think let's merge this asap, just after the remaining small changes
Deleting:
- .Rhistory
- man/figures/README-pressure-1.png

**Comment by WillemVervoort** [comment]
All done, please merge

**Pull request merged**